lib/network/rest/server.js

```diff
@@ -0,0 +1,332 @@
+'use strict'; // eslint-disable-line
+
+const httpServer = require('../http/server');
+const Logger = require('werelogs').Logger;
+const url = require('url');
+const assert = require('assert');
+
+const constants = require('../../constants');
+const utils = require('./utils');
+const errors = require('../../errors');
+
+/**
+ * Parse the Range header into a two-integers array with possibly
+ * undefined values
+ *
+ * @param {String} rangeHeader - The 'Range' header value
+ * @return {(Number|undefined) []} An array of [left, right] integer
+ *   values provided in the range specification if present, or
+ *   undefined otherwise.
+ */
+function parseRangeSpec(rangeHeader) {
+    const rangeMatch = /^bytes=([0-9]+)?-([0-9]+)?$/.exec(rangeHeader);
+    // ignore header if syntax is invalid
+    if (rangeMatch) {
+        return rangeMatch.slice(1, 3).map(
+            str => (str === undefined ?
+                    undefined : Number.parseInt(str, 10)));
+    }
+    return undefined;
```

Comment:
It would allow to log something when catching it, indeed. But in order to be HTTP compliant (we not necessarily have to for a local daemon) we have to catch the exception and return a 200 OK anyway.